Week3/homework/step2-6.js

```diff
@@ -4,13 +4,25 @@ const arr2d = [[1, 2], [3, 4], [5, 6]];
 const arr3d = [[[1, 2], [3, 4]], [[5, 6], [7, 8]]];
 
 function flattenArray2d(arr) {
-  // Replace this comment and the next line with your code
-  console.log(arr);
+  const flattenedArray = [];
+  for (let i = 0; i < arr.length; i++) {
+    for (let j = 0; j < arr[i].length; j++) {
+      flattenedArray.push(arr[i][j]);
+    }
+  }
+  return flattenedArray;
 }
 
 function flattenArray3d(arr) {
-  // Replace this comment and the next line with your code
-  console.log(arr);
+  const flattened = [];
+  for (let i = 0; i < arr.length; i++) {
+    for (let j = 0; j < arr[i].length; j++) {
+      for (let k = 0; k < arr[i][j].length; k++) {
+        flattened.push(arr[i][j][k]);
```

Comment:
Excellent job, this ran fine which is doing the task.
But sometimes we don't only have to do the task, but to think of `what if my program have this situation`?
```javascript
// this will not be as you expected!
// but it will result with:  [ [ [ 1, 2 ], [ 3, 4 ] ], [ [ 5, 6 ], [ 7, 8 ] ] ] -> half way 😅
const arrXd = [[[[[1, 2], [3, 4]], [[5, 6], [7, 8]]]]];
```

you can scroll [here] to see how people solved this issue with many solutions, but for simplicity I would go for the first [result on google]

[here]: https://stackoverflow.com/questions/10865025/merge-flatten-an-array-of-arrays
[result on google]: https://www.google.com/search?sxsrf=ACYBGNRxR8JYJ8_YygPAGLO693S5jrVJhQ%3A1569024533993&ei=FWqFXYalPIfYwQKgk6SIBg&q=flatten+arrays+with+javascript+mdn&oq=flatten+arrays+with+javascript+mdn&gs_l=psy-ab.3..33i22i29i30.163590.164533..164837...0.2..0.88.316.4......0....1..gws-wiz.......0i71j0i22i30.qPtLYprnxbc&ved=0ahUKEwjGg_jvz-DkAhUHbFAKHaAJCWEQ4dUDCAs&uact=5

Reply:
Now, i read the links and learned about `.isArray();` and `.flat(Infinity);`  functions. Thanks a lot